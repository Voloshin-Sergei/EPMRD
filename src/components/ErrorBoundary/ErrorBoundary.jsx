import React from 'react';
import propTypes from 'prop-types';
import style from './ErrorBoundary.module.scss';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    const { hasError } = this.state;
    const { children } = this.props;

    if (hasError) {
      return (
        <div className={style.container}>
          <p className={style.text}>Something wrong :(</p>
        </div>
      );
    }
    return children;
  }
}

ErrorBoundary.propTypes = {
  children: propTypes.element.isRequired,
};

export default ErrorBoundary;
