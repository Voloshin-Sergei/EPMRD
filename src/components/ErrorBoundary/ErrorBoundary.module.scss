.error {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.error p {
  font-size: 36px;
}
