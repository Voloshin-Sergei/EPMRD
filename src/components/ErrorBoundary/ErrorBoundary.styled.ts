import styled from 'styled-components';

export const StyledContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
`;
export const StyledError = styled.p`
  font-size: 3.6rem;
`;
