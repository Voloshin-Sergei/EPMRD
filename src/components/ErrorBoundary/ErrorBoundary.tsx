import React from 'react';
import style from './ErrorBoundary.module.scss';

export interface ErrorBoundaryProps {
  children: React.ReactNode;
}

export interface ErrorBoundaryState {
  hasError: boolean;
}
export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch(): void {
    this.setState({ hasError: true });
  }

  render(): React.ReactNode {
    const { hasError } = this.state;
    const { children } = this.props;

    if (hasError) {
      return (
        <div className={style.container}>
          <p className={style.text}>Something wrong :(</p>
        </div>
      );
    }
    return children;
  }
}
