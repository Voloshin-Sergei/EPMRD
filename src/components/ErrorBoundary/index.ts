export { ErrorBoundary, ErrorBoundaryProps, ErrorBoundaryState } from './ErrorBoundary';
