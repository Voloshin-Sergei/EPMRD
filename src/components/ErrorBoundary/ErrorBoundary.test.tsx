import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { ErrorBoundary }  from './ErrorBoundary';
Enzyme.configure({ adapter: new Adapter() });

const Something = () => null;

describe("ErrorBoundary component", () => {
    it('should render ErrorBoundary component', () => {
      const wrapper = shallow(
        <ErrorBoundary>
          <Something />
        </ErrorBoundary>
      );
      const error = new Error('test');
      wrapper.find(Something).simulateError(error);
      expect(wrapper).toMatchSnapshot();
  });
});