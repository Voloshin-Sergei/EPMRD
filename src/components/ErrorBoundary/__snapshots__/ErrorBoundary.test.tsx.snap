// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should render ErrorBoundary component 1`] = `
<div
  className="container"
>
  <p
    className="text"
  >
    Something wrong :(
  </p>
</div>
`;
